app/code/Magento/BundleGraphQl/Model/Resolver/Options/Label.php

```diff
@@ -7,66 +7,50 @@
 
 namespace Magento\BundleGraphQl\Model\Resolver\Options;
 
+use Magento\Framework\GraphQl\Exception\GraphQlInputException;
 use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
-use Magento\CatalogGraphQl\Model\Resolver\Products\DataProvider\Deferred\Product;
+use Magento\CatalogGraphQl\Model\Resolver\Products\DataProvider\Deferred\Product as ProductDataProvider;
 use Magento\Framework\GraphQl\Config\Element\Field;
-use Magento\Framework\GraphQl\Query\Resolver\Value;
-use Magento\Framework\GraphQl\Query\Resolver\ValueFactory;
 use Magento\Framework\GraphQl\Query\ResolverInterface;
 
 /**
  * Class Label
  */
 class Label implements ResolverInterface
 {
-
     /**
-     * @var ValueFactory
-     */
-    private $valueFactory;
-
-    /**
-     * @var Product
+     * @var ProductDataProvider
      */
     private $product;
 
     /**
-     * @param ValueFactory $valueFactory
-     * @param Product $product
+     * @param ProductDataProvider $product
      */
-    public function __construct(ValueFactory $valueFactory, Product $product)
+    public function __construct(ProductDataProvider $product)
     {
-        $this->valueFactory = $valueFactory;
         $this->product = $product;
     }
 
     /**
-     * @inheritDoc
+     * @inheritdoc
      */
     public function resolve(
         Field $field,
         $context,
         ResolveInfo $info,
         array $value = null,
         array $args = null
-    ): Value {
+    ) {
         if (!isset($value['sku'])) {
-            $result = function () {
-                return null;
-            };
-            return $this->valueFactory->create($result);
+            throw new GraphQlInputException(__('"sku" value should be specified'));
         }
 
         $this->product->addProductSku($value['sku']);
         $this->product->addEavAttributes(['name']);
 
-        $result = function () use ($value) {
-            $productData = $this->product->getProductBySku($value['sku']);
-            /** @var \Magento\Catalog\Model\Product $productModel */
-            $productModel = isset($productData['model']) ? $productData['model'] : null;
-            return $productModel ? $productModel->getName() : null;
-        };
-
-        return $this->valueFactory->create($result);
+        $productData = $this->product->getProductBySku($value['sku']);
```

Comment:
This was not supposed to be replaced because this was actual performance optimization via Deferred mechanism.

Reply:
Fixed